include/GafferSceneUI/Private/OutputBuffer.h

```diff
@@ -81,6 +81,9 @@ class OutputBuffer
 		using BufferChangedSignal = Gaffer::Signals::Signal<void()>;
 		BufferChangedSignal &bufferChangedSignal();
 
+		const Imath::Box2i &dataWindow() const;
+		const std::vector<float> &rgbaBuffer() const;
```

Comment:
Exposing these directly is dangerous, because the buffers can be modified from a background thread at any time - see `OutputBuffers::m_bufferReallocationMutex`. Unless we have other uses for exposing the buffers directly, and a mechanism for protecting that, I think I'd suggest we replace them with a `BufferAlgo::snapshotToFile()` method which we forward to directly from `SceneGadget::snapshotToFile()`.

Reply:
I removed those in https://github.com/GafferHQ/gaffer/pull/5533/commits/5dead3ccef2b62c176f6f43e542c7f16794a6df6 and shifted the implementation to `OutputBuffer::snapshotToFile()` in that commit and https://github.com/GafferHQ/gaffer/pull/5533/commits/b6c2f5367845b1fe6f4d802a0ea795e2c95c9fa6. Should it actually be in `BufferAlgo::snapshotToFile()`? `BufferAlgo` seems more about image window tools than handling the buffer lock and such, so I put it in `OutputBuffer` for now.

Reply:
Sorry, yep, I meant `OutputBuffer` rather than `BufferAlgo` - thanks for translating my gibberish into something sensible.